crates/glaredb/src/rpc_proxy.rs

```diff
@@ -28,11 +33,33 @@ impl RpcProxy {
         // This _may_ end up killing inflight queries, but we can handle that
         // later.
 
-        Server::builder()
-            .trace_fn(|_| debug_span!("rpc_proxy_service_request"))
-            .add_service(ExecutionServiceServer::new(self.handler))
-            .serve(addr)
-            .await?;
+        let mut server = Server::builder().trace_fn(|_| debug_span!("rpc_proxy_service_request"));
+
+        match (rpc_server_cert, rpc_server_key) {
+            (Some(cert), Some(key)) => {
+                let cert = std::fs::read_to_string(cert)?;
+                let key = std::fs::read_to_string(key)?;
+                let identity = Identity::from_pem(cert, key);
```

Comment:
looks like here's the place where we're pulling information (like domain?) out of the server's certificate, I'm inclined to think that the client certs would work the same way.

Reply:
Identity just stores the cert and the key. Doesn't provide any other information.

Reply:
https://docs.rs/x509-parser/latest/x509_parser/certificate/struct.X509Certificate.html <- looks like this will pull the info out of the certificate.